.github/workflows/ci.yml

```diff
@@ -123,8 +123,12 @@ jobs:
         - name: Build controller logs
           if: ${{ failure() }}
           run: |
+            POD_NAME=$(kubectl -n shipwright-build get pod --output name)
+            if [ -z "${POD_NAME}" ]; then
+              echo "# Pod is missing, there are no logs to retrieve, bailing out..."
+              exit 1
+            fi
             echo "# Pods:"
-            kubectl -n shipwright-build get pod
             PODS=$(kubectl -n shipwright-build get pod -o json)
             POD_NAME=$(echo "${PODS}" | jq -r '.items[] | select(.metadata.name | startswith("shipwright-build-controller-")) | .metadata.name')
```

Comment:
Hm, not sure if I completely understand in which state you assume to be here.

What one needs to take care is that the e2e tests also use the `shipwright-build` namespace -> Build and BuildRuns and TaskRuns and **TaskRun pods** also are there (unless the refactoring of Matthias changes this). So, even if the build controller pod is gone, there might be other pods. That's why I have the `select(.metadata.name | startswith("shipwright-build-controller-"))` in my `jq` statement.

May you quickly summarize what you try to reach. Maybe we need to do this slightly different.

My print out of the pods that you removed was actually desired so that one can see the restart count.

Reply:
![image](https://user-images.githubusercontent.com/4812480/109835573-9a669800-7c43-11eb-9064-61115286b6e1.png)

If there are none pods in the namespace, it will try to get logs and eventually fail. The output is confusing, so this PR ensures that if this is the case we have a clear message and stop any attempt on retrieving logs.

Reply:
I guess this would work:

```yaml
        - name: Build controller logs
          if: ${{ failure() }}
          run: |
            echo "# Pods:"
            kubectl -n shipwright-build get pod
            PODS=$(kubectl -n shipwright-build get pod -o json)
            POD_NAME=$(echo "${PODS}" | jq -r '.items[] | select(.metadata.name | startswith("shipwright-build-controller-")) | .metadata.name')
            if [ "${POD_NAME}" != "" ]; then
              RESTART_COUNT=$(echo "${PODS}" | jq -r ".items[] | select(.metadata.name == \"${POD_NAME}\") | .status.containerStatuses[0].restartCount")
              if [ "${RESTART_COUNT}" != "0" ]; then
                echo "# Previous logs:"
                kubectl -n shipwright-build logs "${POD_NAME}" --previous || true
              fi
              echo "# Logs:"
              kubectl -n shipwright-build logs "${POD_NAME}"
            fi
```

Reply:
yes, but I think is better to highlight somehow that there are none logs to retrieve. Can you suggest that in the PR so I can add it, or modify mine, as you think is better.

Reply:
```yaml
        - name: Build controller logs
          if: ${{ failure() }}
          run: |
            echo "# Pods:"
            kubectl -n shipwright-build get pod
            PODS=$(kubectl -n shipwright-build get pod -o json)
            POD_NAME=$(echo "${PODS}" | jq -r '.items[] | select(.metadata.name | startswith("shipwright-build-controller-")) | .metadata.name')
            if [ "${POD_NAME}" != "" ]; then
              RESTART_COUNT=$(echo "${PODS}" | jq -r ".items[] | select(.metadata.name == \"${POD_NAME}\") | .status.containerStatuses[0].restartCount")
              if [ "${RESTART_COUNT}" != "0" ]; then
                echo "# Previous logs:"
                kubectl -n shipwright-build logs "${POD_NAME}" --previous || true
              fi
              echo "# Logs:"
              kubectl -n shipwright-build logs "${POD_NAME}"
            else
              echo "# Pod is missing, there are no logs to retrieve, bailing out..."
            fi
```

Reply:
done, please resolve